dll/win32/browseui/internettoolbar.cpp

```diff
@@ -1504,8 +1504,14 @@ HRESULT STDMETHODCALLTYPE CInternetToolbar::AddBand(IUnknown *punk)
 
 HRESULT STDMETHODCALLTYPE CInternetToolbar::EnumBands(UINT uBand, DWORD *pdwBandID)
 {
-    UNIMPLEMENTED;
-    return E_NOTIMPL;
+    if (uBand == ~0ul)
+        return ::SendMessage(fMainReBar, RB_GETBANDCOUNT, 0, 0);
+    int id;
+    IUnknown *pUnk;
+    HRESULT hr = EnumBands(uBand, &id, &pUnk);
```

Comment:
The `pUnk` doesn't need to be dereferenced or something?

Reply:
No, it does not care about `pUnk` but that parameter is not optional.